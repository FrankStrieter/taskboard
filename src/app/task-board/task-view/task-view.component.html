<h3 name="task-title">{{ task.title }}</h3>
<p name="task-text">{{ task.text }}</p>
