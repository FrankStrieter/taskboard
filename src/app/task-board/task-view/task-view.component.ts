import { Component, Input } from '@angular/core';
import { Task } from '../models';

@Component({
  selector: 'tb-task-view',
  templateUrl: './task-view.component.html',
  styleUrls: ['./task-view.component.scss']
})
export class TaskViewComponent {
  @Input() task: Task;
}
