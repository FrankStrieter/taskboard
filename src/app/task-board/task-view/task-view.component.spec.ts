import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { TaskViewComponent } from './task-view.component';

describe('<tb-task-view>', () => {
  let sut: ComponentFixture<TaskViewComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TaskViewComponent]
    }).compileComponents();

    sut = TestBed.createComponent(TaskViewComponent);

    sut.componentInstance.task = {
      title: 'Buy Milk',
      text: 'Norma Tennenlohe'
    };

    sut.detectChanges();
  }));

  describe('When it is created', () => {
    it('should display the title of the task', () => {
      const title = sut.debugElement.query(By.css('[name=task-title]'))
        .nativeElement;

      expect(title).toMatchSnapshot();
    });

    it('should display the description of the task', () => {
      const text = sut.debugElement.query(By.css('[name=task-text]'))
        .nativeElement;

      expect(text).toMatchSnapshot();
    });
  });
});
