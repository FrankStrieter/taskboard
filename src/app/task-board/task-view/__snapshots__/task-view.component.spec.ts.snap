// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<tb-task-view> When it is created should display the description of the task 1`] = `
<p
  name="task-text"
>
  Norma Tennenlohe
</p>
`;

exports[`<tb-task-view> When it is created should display the title of the task 1`] = `
<h3
  name="task-title"
>
  Buy Milk
</h3>
`;
